charts/argo-cd/templates/argocd-repo-server/deployment.yaml

```
@@ -39,6 +39,9 @@ spec:
 {{- toYaml .Values.repoServer.podLabels | nindent 8 }}
         {{- end }}
     spec:
+      {{- if .Values.repoServer.shareProcessNamespace }}
+      shareProcessNamespace: true
+      {{- end }}
```

Please change the invocation of the template to `with` so it's easier to template and future proof.
```suggestion
      {{- with .Values.repoServer.shareProcessNamespace }}
      shareProcessNamespace: {{ . }}
     {{- end }}
```